警告抑制 : 非静的メンバーをコンストラクタの初期化リストで初期化する

# 目的 : Must

警告を抑制する
## 背景

非静的メンバーをヘッダーで初期化している為、警告が多く出力される。

> warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]

``` shell
In file included from ../hrp2/sdk/workspace/hackEV/instances.h:8:0,
                 from ../hrp2/sdk/workspace/hackEV/SpeedCalculator.cpp:6:
../hrp2/sdk/workspace/hackEV/Logger.h:59:38: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     const char*  LOGDIRECTORY_PATH = "/ev3rt/logs";
                                      ^
../hrp2/sdk/workspace/hackEV/Logger.h:62:33: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     const char*  LOGFILE_NAME = "hackEV_log.csv";
                                 ^
In file included from ../hrp2/sdk/workspace/hackEV/instances.h:9:0,
                 from ../hrp2/sdk/workspace/hackEV/SpeedCalculator.cpp:6:
../hrp2/sdk/workspace/hackEV/DriveController.h:49:29: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     float   directionLast = 0.0F;
                             ^
../hrp2/sdk/workspace/hackEV/DriveController.h:52:30: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     float   directionTotal = 0.0F;
                              ^
../hrp2/sdk/workspace/hackEV/DriveController.h:55:28: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     float   distanceLast = 0.0F;
                            ^
../hrp2/sdk/workspace/hackEV/DriveController.h:58:29: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     float   distanceTotal = 0.0F;
                             ^
../hrp2/sdk/workspace/hackEV/DriveController.h:61:32: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     float lastGetDistanceLeft =0.0F;
                                ^
../hrp2/sdk/workspace/hackEV/DriveController.h:64:33: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     float lastGetDistanceRight =0.0F;
                                 ^
../hrp2/sdk/workspace/hackEV/DriveController.h:67:25: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     int lastPowerLeft = 0;
                         ^
../hrp2/sdk/workspace/hackEV/DriveController.h:70:26: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     int lastPowerRight = 0;
                          ^
../hrp2/sdk/workspace/hackEV/DriveController.h:73:23: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     SYSTIM lastTime = 0;
                       ^
../hrp2/sdk/workspace/hackEV/DriveController.h:76:23: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     SYSTIM DURATION = 100;
                       ^
../hrp2/sdk/workspace/hackEV/DriveController.h:79:31: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     float OnePowerDeviation = 0.084107F;
                               ^
In file included from ../hrp2/sdk/workspace/hackEV/instances.h:10:0,
                 from ../hrp2/sdk/workspace/hackEV/SpeedCalculator.cpp:6:
../hrp2/sdk/workspace/hackEV/sensors/GyroSensorController.h:39:41: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11 [enabled by default]
     const int16_t STAGE_ON_THRESHOLD = -90;
                                         ^
```
# やる事 : Must
- [ ] 非静的メンバーをコンストラクタの初期化リストで初期化する
# 考慮した事
## [リスク](https://kotobank.jp/word/%E3%83%AA%E3%82%B9%E3%82%AF-183705)

なし
## [懸念事項](https://kotobank.jp/word/%E6%87%B8%E5%BF%B5-490895)

なし
# 関連項目
## 資料

[Non-static data member initializers - しるふぃずむ](http://yuhsylphy.hateblo.jp/entry/20130608/1370662892)

> 非静的メンバを構築時に初期化するには，コンストラクタにメンバ初期化子を記述する必要がありました．
> C++11では，静的メンバの初期化子に近い形で直感的に非静的メンバの初期化子を記述することができるようになりました．

--- a/hrp2/sdk/workspace/hackEV/DriveController.cpp
+++ b/hrp2/sdk/workspace/hackEV/DriveController.cpp
@@ -30,6 +30,9 @@ DriveController::DriveController()
     , lastGetDistanceRight(0.0F)
     , DURATION(100.0F)
     , OnePowerDeviation(0.084107F)
+    , lastPowerLeft(0)
+    , lastPowerRight(0)
+    , lastTime(0)
 {
 }
 

--- a/hrp2/sdk/workspace/hackEV/DriveController.h
+++ b/hrp2/sdk/workspace/hackEV/DriveController.h
@@ -46,37 +46,37 @@ private:
     MotorWheel  *motorWheelRight;
 
     //! 前回の向きの累積[単位 : 度]
-    float   directionLast = 0.0F;
+    float   directionLast;
 
     //! 向きの累積[単位 : 度]
-    float   directionTotal = 0.0F;
+    float   directionTotal;
 
     //! 前回の距離の累積[単位 : cm]
-    float   distanceLast = 0.0F;
+    float   distanceLast;
 
     //! 距離の累積[単位 : cm]
-    float   distanceTotal = 0.0F;
+    float   distanceTotal;
     
     //! 最後に左タイヤの距離を取得したときの値
-    float lastGetDistanceLeft =0.0F;
+    float lastGetDistanceLeft;
     
     //! 最後に右タイヤの距離を取得したときの値
-    float lastGetDistanceRight =0.0F;
+    float lastGetDistanceRight;
     
     //! 最後に左タイヤに設定した出力値
-    int lastPowerLeft = 0;
+    int lastPowerLeft;
     
     //! 最後に右タイヤに設定した出力値
-    int lastPowerRight = 0;
+    int lastPowerRight;
     
     //! 時間間隔測定に用いるための最後に取得した時間
-    SYSTIM lastTime = 0;
+    SYSTIM lastTime;
     
     //! 出力値を補正する時間間隔
-    SYSTIM DURATION = 100;
+    SYSTIM DURATION;
     
     //! 1パワー分の100ms間に走行する距離(値はモデル図のグラフから平均を算出)
-    float OnePowerDeviation = 0.084107F;
+    float OnePowerDeviation;
 
     //! \addtogroup 速度測定用のインスタンス
     //@{

--- a/hrp2/sdk/workspace/hackEV/Logger.cpp
+++ b/hrp2/sdk/workspace/hackEV/Logger.cpp
@@ -13,6 +13,7 @@
 */
 Logger::Logger()
     : fpLog(NULL)
+    , LOGFILE_NAME("hackEV_log.csv")
     , loggerInfo(NULL)
     , enabled(false)
     , outputHeader(false)

--- a/hrp2/sdk/workspace/hackEV/Logger.h
+++ b/hrp2/sdk/workspace/hackEV/Logger.h
@@ -55,11 +55,8 @@ private:
     //! ログファイルのファイルポインタ
     FILE    *fpLog;
 
-    //! ログファイルを収めるディレクトリ
-    const char*  LOGDIRECTORY_PATH = "/ev3rt/logs";
-
     //! ログファイルのファイル名
-    const char*  LOGFILE_NAME = "hackEV_log.csv";
+    char*  LOGFILE_NAME;
 
     //! 蓄積するログ情報
     vector<USER_LOG> loggerInfo;

--- a/hrp2/sdk/workspace/hackEV/sensors/GyroSensorController.cpp
+++ b/hrp2/sdk/workspace/hackEV/sensors/GyroSensorController.cpp
@@ -12,6 +12,7 @@
 //! class for gyro sensor
 GyroSensorController::GyroSensorController(sensor_port_t _port)
     : port(_port)
+    , STAGE_ON_THRESHOLD(-90)
 {
 }
 

--- a/hrp2/sdk/workspace/hackEV/sensors/GyroSensorController.h
+++ b/hrp2/sdk/workspace/hackEV/sensors/GyroSensorController.h
@@ -36,6 +36,5 @@ private:
     int16_t gyroRate;
 
     //! 段差乗り上げの際に、ジャイロセンサーが受け取る角速度[度/秒]の閾値。(ただし、約25ms間隔で計測した場合のもの。)
-    const int16_t STAGE_ON_THRESHOLD = -90;
-
+    const int16_t STAGE_ON_THRESHOLD;
 };

